rocAL/rocAL_pybind/CMakeLists.txt

```diff
@@ -47,66 +50,138 @@ if(GPU_SUPPORT)
+    if(NOT HIP_FOUND)
+        set(BUILD_ROCAL_PYBIND false)
+        message("-- ${Yellow}rocAL pybind requires HIP, Not Found${ColourReset}")
+    endif()
+    if(NOT AMDRPP_FOUND)
+        set(BUILD_ROCAL_PYBIND false)
+        message("-- ${Yellow}NOTE: rocAL pybind requires AMDRPP, Not Found${ColourReset}")
+    else()
+        #find the RPP backend type
+        set(RPP_BACKEND_OPENCL_FOUND 0)
+        set(RPP_BACKEND_HIP_FOUND 0)
+        if(EXISTS ${AMDRPP_INCLUDE_DIRS}/rpp_backend.h)
+            file(READ ${AMDRPP_INCLUDE_DIRS}/rpp_backend.h RPP_BACKEND_FILE)
+            string(REGEX MATCH "RPP_BACKEND_OPENCL ([0-9]*)" _ ${RPP_BACKEND_FILE})
+            set(RPP_BACKEND_OPENCL_FOUND ${CMAKE_MATCH_1})
+            string(REGEX MATCH "RPP_BACKEND_HIP ([0-9]*)" _ ${RPP_BACKEND_FILE})
+            set(RPP_BACKEND_HIP_FOUND ${CMAKE_MATCH_1})
+        else()
+            message("-- ${Red}WARNING: ${AMDRPP_INCLUDE_DIRS}/rpp_backend.h file Not Found. please run the setup script to install latest RPP package ${ColourReset}")
         endif()
-    elseif("${BACKEND}" STREQUAL "HIP" AND HIP_FOUND)
-        if (NOT RPP_BACKEND_HIP_FOUND)
+        if ("${BACKEND}" STREQUAL "OPENCL" AND OpenCL_FOUND)
+            if (NOT RPP_BACKEND_OPENCL_FOUND)
+                set(BUILD_ROCAL_PYBIND false)
+                message("-- ${Red}WARNING: RPP with OpenCL backend Not Found (HINT: run setup script with the OCL backend first!) -- rocAL pybind module excluded${ColourReset}")
+            endif()
+        elseif("${BACKEND}" STREQUAL "HIP" AND HIP_FOUND)
+            if (NOT RPP_BACKEND_HIP_FOUND)
+                set(BUILD_ROCAL_PYBIND false)
+                message("-- ${Red}WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- rocAL pybind module excluded${ColourReset}")
+            endif()
+        else()
             set(BUILD_ROCAL_PYBIND false)
-            message("-- ${Red}WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- rocAL pybind module excluded${ColourReset}")
+            message("-- ${Red}WARNING: GPU support with OpenCL or HIP Not Found -- rocAL Pybind module excluded${ColourReset}")
         endif()
-    else()
+    endif()
+
+    if(NOT TurboJpeg_FOUND)
         set(BUILD_ROCAL_PYBIND false)
-        message("-- ${Red}WARNING: GPU support with OpenCL or HIP Not Found -- rocAL Pybind module excluded${ColourReset}")
+        message("-- ${Yellow}NOTE: rocAL pybind requires TurboJpeg, Not Found${ColourReset}")
     endif()
-endif()
 
-if(NOT TurboJpeg_FOUND)
-    set(BUILD_ROCAL_PYBIND false)
-    message("-- ${Yellow}NOTE: rocAL pybind requires TurboJpeg, Not Found${ColourReset}")
-endif()
 
-if(${BUILD_ROCAL_PYBIND})
+    if(${BUILD_ROCAL_PYBIND})
 
-    link_directories(${ROCM_PATH}/lib)
+        find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
```

Comment:
move the find_package(Python3 from here to outside of the if(${BUILD_ROCAL_PYBIND}). 
add an if(NOT Python3_FOUND) check after find_package(Python3 code and make sure to set the BUILD_ROCAL_PYBIND to false as Python3 is a dependency.